runtime/src/bank.rs

```diff
@@ -1775,12 +1803,19 @@ impl Bank {
         let mut partitions = vec![];
         if parent_epoch < current_epoch {
             if current_slot_index > 0 {
+                // generate and push gapped partitions because some slots are skipped
                 let parent_last_slot_index = self.get_slots_in_epoch(parent_epoch) - 1;
+
+                // ... for parent epoch
                 partitions.push(self.partition_from_slot_indexes(
                     parent_slot_index,
                     parent_last_slot_index,
                     parent_epoch,
+                    true,
                 ));
+
+                // ... for current epoch
+                partitions.push(self.partition_from_slot_indexes(0, 0, current_epoch, true));
```

Comment:
Fortunately, gating is simple. I'll just surround this `partitions.push(...)` with `if self.operating_mode() == OOO && self.epoch() > XX` for devnet, testnet, mainnet-beta respectively.